### PR TITLE
feat: add docker layer caching support on machine executors

### PR DESCRIPTION
**Pull request by a-cordier**
see https://github.com/CircleCI-Public/circleci-config-sdk-ts/issues/146

### CODE REVIEW

**Comment by KyleTryon** [comment]
Hey @a-cordier Thanks for the contribution!
I see here the Docker Layer Caching boolean was added to the machine executor type.

Docker Layer Caching is actually available on the Docker Container as well: https://circleci.com/docs/docker-layer-caching

We _probably_ could do some very advanced typescript stuff to make it so that you require the `setup_remote_docker` step on the docker-based job implementing DLC, but i think for now we could potentially get away with adding the boolean  to the parent `Executor` class and extend it down to both.

CC @Jaryt for opinion as well.

**Comment by a-cordier** [comment]
@KyleTryon Thanks for the prompt review ! Does the docker_layer_caching property applies to (MacOS|Windows) executors ? I can see that they both extend the Executor class.

**Comment by a-cordier** [comment]
Just figured out that dlc is implemented with a command for docker executors.

I've updated the PR to add dlc support on the command as well

**Review by KyleTryon — COMMENTED** [review]
Well done 👏 . And thank you for looking into the config reference to resolve this. I had to double check that myself. I am starting the tests now.

edit: oops, just noticed the branch is out of date. Going to update that first, fingers crossed for no conflicts

File: src/lib/Components/Commands/exports/Native/SetupRemoteDocker.ts [hunk 1]
```diff
@@ -31,6 +31,17 @@ export class SetupRemoteDocker implements Command {
   get generableType(): GenerableType {
     return GenerableType.SETUP_REMOTE_DOCKER;
   }
+
+  /**
+   * Enable docker image layer caching
+   * @param {boolean} enabled - If true, docker layer caching is enabled for the job.
```
Comment (KyleTryon) [inline src/lib/Components/Commands/exports/Native/SetupRemoteDocker.ts:37]:
```suggestion
   * @param enabled - If true, docker layer caching is enabled for the job.
```

In typescript, the type is automatically inferred here.

File: src/lib/Components/Commands/exports/Native/SetupRemoteDocker.ts [hunk 2]
```diff
@@ -31,6 +31,17 @@ export class SetupRemoteDocker implements Command {
...
+   * @returns {SetupRemoteDocker} - The current instance of the SetupRemoteDocker Command.
```
Comment (KyleTryon) [inline src/lib/Components/Commands/exports/Native/SetupRemoteDocker.ts:38]:
```suggestion
   * @returns SetupRemoteDocker - The current instance of the SetupRemoteDocker Command.
```

**Review by KyleTryon — CHANGES_REQUESTED** [review]
This should resolve the linting errors. Minor differences between JSDoc and TSDoc.
https://app.circleci.com/pipelines/github/CircleCI-Public/circleci-config-sdk-ts/439/workflows/7a9acf7c-28ec-410e-9a49-46888efbb375/jobs/2299

File: src/lib/Components/Executors/exports/MachineExecutor.ts [hunk 3]
```diff
@@ -33,4 +42,15 @@ export class MachineExecutor extends Executor<MachineResourceClass> {
   get executorLiteral(): ExecutorLiteral {
     return 'machine';
   }
+
+  /**
+   * Enable docker image layer caching
+   * @param {boolean} enabled - If true, docker layer caching is enabled for the machine executor.
```
Comment (KyleTryon) [inline src/lib/Components/Executors/exports/MachineExecutor.ts:48]:
```suggestion
   * @param enabled - If true, docker layer caching is enabled for the machine executor.
```

**Pull request merged**